pkg/policy/execute_test.go

```diff
@@ -712,8 +712,9 @@ func TestGenerateExecutionResultFile(t *testing.T) {
 
 	for _, tc := range testCases {
 		executiionResult := ExecutionResult{PolicyName: tc.PolicyName}
-		err := GenerateExecutionResultFile(&executiionResult, tc.OutputDir)
-		if !assert.NoError(t, err) {
+		diags := GenerateExecutionResultFile(&executiionResult, tc.OutputDir)
+
+		if !assert.False(t, diags.HasErrors()) {
```

Comment:
nit: just use `assert.True` instead of `!assert.False` ?

Reply:
I don't believe that's right :)
I want to assert that there are no errors (i.e. `assert.False(diags.HasErrors())`).
But if there are errors (i.e. `HasErrors` is `true`), I want to `continue` to the next testcase.
An assertion returns "true" if the condition specified is true, and false otherwise (i.e. it will return false if there are errors in diags)..

Reply:
Will:
```go
if diags.HasErrors() {
	assert.Fail(t, "expected no errors, got %v", diags)
	continue
}
```

work?
That way we'll get the received errors in the message

Reply:
done :)